app/src/main/java/it/niedermann/nextcloud/deck/ui/appwidget/StackWidget.java

```diff
@@ -0,0 +1,99 @@
+package it.niedermann.nextcloud.deck.ui.appwidget;
+
+import android.app.PendingIntent;
+import android.appwidget.AppWidgetManager;
+import android.appwidget.AppWidgetProvider;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.widget.RemoteViews;
+
+import java.util.NoSuchElementException;
+
+import it.niedermann.nextcloud.deck.R;
+import it.niedermann.nextcloud.deck.model.appwidgets.StackWidgetModel;
+import it.niedermann.nextcloud.deck.persistence.sync.SyncManager;
+import it.niedermann.nextcloud.deck.ui.MainActivity;
+import it.niedermann.nextcloud.deck.ui.card.EditActivity;
+
+import static android.appwidget.AppWidgetManager.ACTION_APPWIDGET_UPDATE;
+
+public class StackWidget extends AppWidgetProvider {
+    public static final String ACCOUNT_ID_KEY = "stack_widget_account_id";
+    public static final String STACK_ID_KEY = "stack_widget_stack_id";
+    private static final int PENDING_INTENT_OPEN_APP_RQ = 0;
+    private static final int PENDING_INTENT_EDIT_CARD_RQ = 1;
+
+    static void updateAppWidget(Context context, AppWidgetManager awm, int[] appWidgetIds) {
+        final SyncManager syncManager = new SyncManager(context);
+
+        for (int appWidgetId : appWidgetIds) {
+            new Thread(() -> {
+                try {
+                    final StackWidgetModel model = syncManager.getStackWidgetModelDirectly(appWidgetId);
+                    RemoteViews views = new RemoteViews(context.getPackageName(), R.layout.widget_stack);
+                    Intent serviceIntent = new Intent(context, StackWidgetService.class);
+
+                    serviceIntent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId);
+                    serviceIntent.putExtra(ACCOUNT_ID_KEY + appWidgetId, model.getAccountId());
+                    serviceIntent.putExtra(STACK_ID_KEY + appWidgetId, model.getStackId());
+                    serviceIntent.setData(Uri.parse(serviceIntent.toUri(Intent.URI_INTENT_SCHEME)));
+
+                    Intent intent = new Intent("android.intent.action.MAIN");
+                    intent.setComponent(new ComponentName(context.getPackageName(), MainActivity.class.getName()));
+                    PendingIntent pendingIntent = PendingIntent.getActivity(context,  PENDING_INTENT_OPEN_APP_RQ,
+                            intent, PendingIntent.FLAG_UPDATE_CURRENT);
+                    views.setOnClickPendingIntent(R.id.widget_stack_header_rl, pendingIntent);
+
+                    Intent templateIntent = new Intent(context, EditActivity.class);
+                    PendingIntent templatePI = PendingIntent.getActivity(context, PENDING_INTENT_EDIT_CARD_RQ,
+                            templateIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+                    views.setPendingIntentTemplate(R.id.stack_widget_lv, templatePI);
+
+                    views.setRemoteAdapter(appWidgetId, R.id.stack_widget_lv, serviceIntent);
```

Comment:
This method is deprecated, are you able to replace it?

Reply:
Sorted now.